src/api.rs

```diff
@@ -30,7 +30,7 @@ use self::sys::responses::WrappingLookupResponse;
 /// are automatically logged accordingly.
 #[derive(Deserialize, Debug)]
 pub struct EndpointResult<T> {
-    pub data: Option<T>,
+    pub data: Option<EitherData<T>>,
```

Comment:
Also worth pointing out explicitly why this was changed:

as it seems that for currently implemented auth methods there is no "data" field returned for login api but for ldap/login it returns `"data": {}` for some reason and that breaks normal deserialisation. This "either data" structure ensures that if unexpectedly data field was returned it still can parse json object.